components/clarinet-deno/index.test.ts

```diff
@@ -0,0 +1,187 @@
+});
+
+Deno.test("expectUint", () => {
+  assertStrictEquals("u42".expectUint(42), 42n);
+
+  assertThrows(() => "42".expectUint(42));
+});
+
+Deno.test("expectPrincicipal", () => {
+  const contract = "'.bns";
+  assertStrictEquals(contract.expectPrincipal(contract), contract);
+  const address = "'ST1PQHQKV0RJXZFY1DGX8MNSNYVE3VGZJSRTPGZGM";
+  assertStrictEquals(address.expectPrincipal(address), address);
+  const both = `${address}.bns`;
+  assertStrictEquals(both.expectPrincipal(both), both);
+
+  const badFormat = ".bns";
+  assertThrows(() => badFormat.expectPrincipal("'.bns"));
+});
+
+Deno.test("expectBuff", () => {
+  const expect = Uint8Array.from([98, 116, 99]);
+  assertEquals("0x627463".expectBuff(expect), expect);
+});
+
+Deno.test("expectList", () => {
+  assertArrayIncludes("[u1, u2, u3]".expectList(), ["u1", "u2", "u3"]);
+});
+
+Deno.test("expectTuple", () => {
+  assertObjectMatch("{id: u0}".expectTuple(), { id: "u0" });
+});
+
+Deno.test("expectPrintEvent", () => {
+  const id = "ST1PQHQKV0RJXZFY1DGX8MNSNYVE3VGZJSRTPGZGM.color-vote";
+  const events = [
+    {
+      type: "contract_event",
+      contract_event: {
+        contract_identifier: id,
+        topic: "print",
+        value: '"ok"',
+      },
+    },
+  ];
+
+  assertObjectMatch(events.expectPrintEvent(id, '"ok"'), {
+    contract_identifier: id,
+    topic: "print",
+    value: '"ok"',
+  });
+});
+
+Deno.test("types.ok", () => {
+  assertStrictEquals(types.ok("2"), "(ok 2)");
+  types.ok("2").expectOk().expectInt(2);
+});
+
+Deno.test("types.err", () => {
+  assertStrictEquals(types.err("u2"), "(err u2)");
+  types.err("u2").expectErr().expectUint(2);
+});
+
+Deno.test("types.some", () => {
+  assertStrictEquals(types.some("true"), "(some true)");
+  types.some("true").expectSome().expectBool(true);
+});
+
+Deno.test("types.none", () => {
+  assertStrictEquals(types.none(), "none");
+  types.none().expectNone();
+});
+
+Deno.test("types.bool", () => {
+  assertStrictEquals(types.bool(true), "true");
+  assertStrictEquals(types.bool(false), "false");
+  types.bool(true).expectBool(true);
+  types.bool(false).expectBool(false);
+});
+
+Deno.test("types.ascii", () => {
+  assertStrictEquals(types.ascii("hello"), '"hello"');
+  types.ascii("hello").expectAscii("hello");
+});
+
+Deno.test("types.utf8", () => {
+  assertStrictEquals(types.utf8("hello"), 'u"hello"');
+  types.utf8("hello").expectUtf8("hello");
+});
+
+Deno.test("types.int", () => {
+  assertStrictEquals(types.int(2), "2");
+  assertStrictEquals(types.int(2n), "2");
+  types.int(2).expectInt(2);
+});
+
+Deno.test("types.uint", () => {
+  assertStrictEquals(types.uint(2), "u2");
+  assertStrictEquals(types.uint(2n), "u2");
+  types.uint(2).expectUint(2);
+});
+
+Deno.test("types.principal", () => {
+  const addr = "ST1SJ3DTE5DN7X54YDH5D64R3BCB6A2AG2ZQ8YPD5";
+  assertStrictEquals(types.principal(addr), `'${addr}`);
+  types.principal(addr).expectPrincipal(`'${addr}`);
```

Comment:
@lgalabru 
Inconsistency here:
`expectPrincipal` expects a leading `'`, but should it?

(Like, `expectUtf8` doesn't expect a leading `u`)

Reply:
Thanks for this tests!
Is this a non issue, or do you want to leave it as is? If former, could we add comment about it in the code? If latter, can we create an issue? thanks!

Reply:
To be explored in #627